lambdas/services/review_processor_service.py

```
@@ -0,0 +1,167 @@
+
+        Args:
+            sqs_message: SQS message record containing file and failure information
+
+        Raises:
+            InvalidMessageException: If message format is invalid or required fields missing
+            S3FileNotFoundException: If file doesn't exist in staging bucket
+            ClientError: For AWS service errors (DynamoDB, S3)
+        """
+        logger.info("Processing review queue message")
+
+        request_context.patient_nhs_no = review_message.nhs_number
+
+        logger.info(f"Processing review for NHS: {review_message.nhs_number} with {len(review_message.files)} files")
+
+        for file in review_message.files:
+           logger.info(f"Processing review file: {file.file_name}")
+           self._verify_file_exists_in_staging(file.file_path)
+
+        review_id = uuid.uuid4().hex
+        files = self._move_files_to_review_bucket(review_message, review_id)
+        document_upload_review = self._build_review_record(review_message, review_id, files)
+
+        self._create_review_record(document_upload_review)
+
+        logger.info(
+            f"Successfully processed review for {review_message.nhs_number}",
+            {"Result": "Review record created and file moved"},
+        )
+
+    def _verify_file_exists_in_staging(self, file_path: str) -> None:
+        """
+        Verify the file exists in the staging bucket.
+
+        Args:
+            file_path: S3 key of the file in staging bucket
+
+        Raises:
+            S3FileNotFoundException: If file does not exist in staging bucket
+        """
+        try:
+            file_exists = self.s3_service.file_exist_on_s3(s3_bucket_name=self.staging_bucket_name, file_key=file_path)
+
+            if not file_exists:
+                raise S3FileNotFoundException(f"File not found in staging bucket: {file_path}")
+
+            logger.info(f"Verified file exists in staging: {file_path}")
+
+        except S3FileNotFoundException:
+            raise
```

Do we want to raise any specific errors here?